### ISSUE TITLE
Proposed Modifications to PR #5827 and Coupling Between P3 and COSP

### ISSUE THREAD

**Issue by yunpengshan2014** [E3SM-Project/E3SM#6735]
Issue Description: 

PR#5827 introduced an unexpected issue affecting the COSP CALIPSO and CloudSat simulator diagnostics. This issue arises from assigning the P3 diagnostic precipitating solid phase particle falling flux to “precip_ice_flux” for diagnostic purposes. Since “precip_ice_flux” is also a required input field and is treated as snow in the CALIPSO and CloudSat simulators, this results in the double-counting of solid phase particles. This is because the E3SMv3 implemented the single-ice category version of P3 as used in SCREAM, where the simulated solid particles are either precipitating or non-precipitating. 

While assigning the P3 diagnostic precipitating solid phase particle falling flux to “precip_ice_flux” is technically correct, it inadvertently leads to double-counting of ice in the CALIPSO and CloudSat simulator's cloud diagnostics. Note that this issue does not impact model simulations.

More discussion is here: https://acme-climate.atlassian.net/wiki/spaces/ATMOS/pages/edit-v2/4654497798?draftShareId=c179bae8-e00d-4754-8faf-0d05f5eb48a2

**Comment by rljacob** [comment]
PR #5827 indicates it was closed without being merged.  Did this code make it to master?

**Comment by yunpengshan2014** [comment]
> PR #5827 indicates it was closed without being merged. Did this code make it to master?

This code has been merged to V3.0.1. So, it should be also in master.

**Comment by crterai** [comment]
@yunpengshan2014 - I think PR #5827  is not the correct PR # and it should be #5891  Would you correct that above?

**Comment by yunpengshan2014** [comment]
> @yunpengshan2014 - I think PR #5827 is not the correct PR # and it should be #5891 Would you correct that above?

Yes, Chris and Rob. #5891 is the correct PR. I have correct it here and also the confluence page. Thanks for capturing it.
